lib/AST/ASTPrinter.cpp

```diff
@@ -907,14 +907,11 @@ class PrintAST : public ASTVisitor<PrintAST> {
         DC = DC->getParent()->getInnermostTypeContext();
```

Comment:
For instance,
`public extension C1 {
  public enum C1Cases : Int {
  case case1
  }
  public class C11 {
    public enum C11Cases : Int {
    case case11
    }
  }
}

public class C2 : C1 {
  public func C2foo() {}
}`
Print `C2` will trigger assertion when walking into `C11Cases`, because its decl context is `C11`, which is still not the correct context.